src/main/java/core/mate/academy/model/Machine.java

```diff
@@ -1,8 +1,14 @@
 package core.mate.academy.model;
 
-public abstract class Machine implements Workable {
+import core.mate.academy.service.MachineProducer;
+
+public abstract class Machine implements Workable , MachineProducer {
     private String name;
     private String color;
+public Machine(String color, String name){
+    this.name = this.name;
+    this.color = this.color;
```

Comment:
The constructor is incorrectly assigning the fields. Instead of `this.name = this.name;` and `this.color = this.color;`, it should be `this.name = name;` and `this.color = color;` to properly initialize the fields with the provided parameters.